### DISCUSSION TITLE
LightningCLI: Passing objects via link_arguments

### DISCUSSION THREAD

**Discussion by lneukom** [Lightning-AI/pytorch-lightning#13539] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi, 

is there a way to pass objects from a **LightningDataModule** to a **LightningModule** in the **LightningCLI** using **link_arguments**? I tried the following

```
class TestModel(LightningModule):

    def __init__(self, scaler: StandardScaler, *args: Any, **kwargs: Any):
        super().__init__(*args, **kwargs)
        print(scaler)


class TestDataModule(LightningDataModule):

    def __init__(self):
        super().__init__()
        self.scaler = StandardScaler()


class TestCLI(LightningCLI):

    def add_arguments_to_parser(self, parser):
        parser.link_arguments('data.scaler', 'model.scaler', apply_on='instantiate')


TestCLI(TestModel, TestDataModule)
```

but got a ` ValueError: Target key "model.scaler" must be for an individual argument`

**Comment by lneukom** [comment]
@mauvilsa posted here as discussed in slack

**Reply by mauvilsa** [reply]
This is a valid use case for `link_arguments` but currently not supported. I will look at what needs to be done to make this work and comment here.

**Comment by mauvilsa** [comment, marked as answer]
The support for this has been implemented in [jsonargparse](https://github.com/omni-us/jsonargparse). There is no need for any change in pytorch-lightning. Just update the package, i.e. `pip install -U jsonargparse`, and your example should work.

**Reply by mauvilsa** [reply]
@lneukom do let me know if you tried it and if it worked as you expected.

**Reply by lneukom** [reply]
Just tested it, works! Thanks for the quick fix!
